src/game.cpp

```diff
@@ -3366,6 +3371,10 @@ void Game::processPlayerInteraction(std::vector<aabb3f> &highlight_boxes,
 			playeritem = mlist->getItem(client->getPlayerItem());
 	}
 
+	if (playeritem.getDefinition(itemdef_manager).name.empty()) { // override the hand
+		if (InventoryList *hlist = local_inventory->getList("hand"))
```

Comment:
IRC discussion http://irc.minetest.net/minetest-dev/2016-11-23#i_4747336
I would also prefer it moved back outside, for clarity.